Breaking: Remove stitchAsyncCode from SentryOption (#2973)
getsentry/sentry-cocoa: CHANGELOG.md

- Create User and Breadcrumb from map (#2820)

### Fixes 

- Improved performance serializing profiling data (#2863)
- Possible crash in Core Data tracking (#2865)
- Ensure the current GPU frame rate is always reported for concurrent transaction profiling metrics (#2929)

### Removed

> - Remove `stitchAsyncCode` from SentryOption (#2973)

Reviewer: `l`:
```suggestion
- Remove experimental `stitchAsyncCode` from SentryOptions (#2973)
```

> The `stitchAsyncCode` experimental option has been removed from `SentryOptions` as it was not behaving in a consistent way. Its behavior was unpredictable and resulted in unexpected errors. We plan to add it back once we fix it.

Reviewer: `l`:
```suggestion
The `stitchAsyncCode` experimental option has been removed from `SentryOptions` as its behavior was unpredictable and sometimes resulted in unexpected errors. We plan to add it back once we fix it, but we don't have an ETA for it.
```
